test/e2e/monitoring_stack_controller_test.go

```diff
@@ -626,10 +649,17 @@ func msNamespaceSelector(labels map[string]string) stackModifier {
 
 func newMonitoringStack(t *testing.T, name string, mods ...stackModifier) *stack.MonitoringStack {
 	ms := &stack.MonitoringStack{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: stack.GroupVersion.String(),
+			Kind:       "MonitoringStack",
+		},
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      name,
 			Namespace: e2eTestNamespace,
 		},
+		Spec: stack.MonitoringStackSpec{
+			ResourceSelector: &metav1.LabelSelector{},
```

Comment:
Isn't it better to use a `stackModifier` to explicitly set the ResourceSelector since the default is nil?

Reply:
Given that the majority of tests expects the empty selector and this is a convenience function I went with this.
Causes less code changes to existing code too.